app/models/db_file.rb

```diff
@@ -32,6 +32,11 @@ def setup(user_id, upload, args = nil)
     self[:created_user_id] = user_id
     self[:updated_user_id] = user_id
     if upload # Uploaded file
+      # check to see it is a valid filename
+      if upload.original_filename.match(/^[a-zA-Z0-9._\-]+$/).nil?
```

Comment:
filenames can only contain letters, numbers, periods and dashes (-)?

Reply:
correct